pandas/core/groupby.py

```diff
@@ -333,6 +335,17 @@ def _set_grouper(self, obj, sort=False):
     def groups(self):
         return self.grouper.groups
 
+    def __repr__(self):
```

Comment:
not sure why you need to introspect at all. when repr is called all of the values are set, simply iterate thru them.

Reply:
It is to get the difference between the default values and the actual value, and only use attrubutes that deviate from the default values.